Services/Common/OJS.Services.Common.Data/Implementations/DataService.cs

```diff
@@ -1,215 +1,226 @@
-            Expression<Func<TEntity, object>>? orderBy = null,
-            bool descending = false,
-            int? skip = null,
-            int? take = null)
+        if (this.ignoreQueryFilters || this.ignoreQueryFiltersOnNextQuery)
         {
-            var query = this.dbSet.AsQueryable();
-
-            if (this.IgnoreQueryFilters)
+            if (this.useQueryFiltersOnNextQuery)
             {
-                query = query.IgnoreQueryFilters();
+                this.useQueryFiltersOnNextQuery = false;
             }
-            else if (this.ignoreQueryFiltersOnNextQuery)
+            else
             {
                 query = query.IgnoreQueryFilters();
                 this.ignoreQueryFiltersOnNextQuery = false;
             }
+        }
 
-            if (filter != null)
-            {
-                query = query.Where(filter);
-            }
-
-            if (orderBy != null)
-            {
-                query = descending
-                    ? query.OrderByDescending(orderBy)
-                    : query.OrderBy(orderBy);
-            }
+        if (filter != null)
+        {
+            query = query.Where(filter);
+        }
 
-            if (skip.HasValue)
-            {
-                query = query.Skip(skip.Value);
-            }
+        if (orderBy != null)
+        {
+            query = descending
+                ? query.OrderByDescending(orderBy)
+                : query.OrderBy(orderBy);
+        }
 
-            if (take.HasValue)
-            {
-                query = query.Take(take.Value);
-            }
+        if (skip.HasValue)
+        {
+            query = query.Skip(skip.Value);
+        }
 
-            return query;
+        if (take.HasValue)
+        {
+            query = query.Take(take.Value);
         }
 
-        public IQueryable<TEntity> GetQueryForUser(
-            UserInfoModel user,
-            Expression<Func<TEntity, bool>>? filter = null,
-            Expression<Func<TEntity, object>>? orderBy = null,
-            bool descending = false,
-            int? skip = null,
-            int? take = null)
-            => this.GetQuery(this.GetUserFilter(user).CombineAndAlso(filter), orderBy, descending, skip, take);
+        return query;
+    }
+
+    public IQueryable<TEntity> GetQueryForUser(
+        UserInfoModel user,
+        Expression<Func<TEntity, bool>>? filter = null,
+        Expression<Func<TEntity, object>>? orderBy = null,
+        bool descending = false,
+        int? skip = null,
+        int? take = null)
+        => this.GetQuery(this.GetUserFilter(user).CombineAndAlso(filter), orderBy, descending, skip, take);
+
+    public IDataService<TEntity> WithQueryFilters()
+    {
+        this.useQueryFiltersOnNextQuery = true;
+        return this;
+    }
+
+    IDataService<TEntity> IDataService<TEntity>.IgnoreQueryFilters()
+    {
+        this.ignoreQueryFiltersOnNextQuery = true;
+        return this;
+    }
```

Comment:
Can you make this public as well and remove the `IDataService<TEntity>.` from IDataService<TEntity>.IgnoreQueryFilters

Reply:
Done.